invoices/invoices.go

```diff
@@ -0,0 +1,786 @@
+package invoices
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb/types"
+	"github.com/lightningnetwork/lnd/feature"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/record"
+)
+
+const (
+	// MaxMemoSize is maximum size of the memo field within invoices stored
+	// in the database.
+	MaxMemoSize = 1024
```

Comment:
Hmm, I think we need to develop a philosophy here w.r.t the types f constants/variables we'll declare here at the interface level, vs within the implementation level (current bolt kv mapping).

Reply:
+1, the comment wasn't updated when I copy&pasted it from `channeldb`. However, it is not clear for me if these limits are domain limits or db. Does it make sense to do not apply them at domain level when the db cannot store them?